StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet.swift

```diff
@@ -13,6 +13,75 @@ import PassKit
 @_spi(STP) import StripePaymentsUI
 import UIKit
 
+protocol IntentAbstraction: PaymentMethodRequirementProvider {
+    /// Extracts all the recommended `PaymentMethodType`s from the given `intent`.
+    /// - Returns: An ordered list of all the `PaymentMethodType`s for this `intent`.
+    var recommendedPaymentSheetMethodTypes: [PaymentSheet.PaymentMethodType] { get }
+    
+    var recommendedPaymentMethodTypes: [STPPaymentMethodType] { get }
+    
+    var isSettingUp: Bool { get }
+    
+    var unactivatedPaymentMethodTypes: [STPPaymentMethodType] { get }
+    
+    var currency: String? { get }
+    
+    var amount: Int? { get }
+    
+    var supportsLink: Bool { get }
```

Comment:
Huh, this is probably something `elements/sessions` returns.

Reminds me that we really should not be putting the `elements/sessions` response into a PaymentIntent/SetupIntent.

Reply:
Actually just a little helper, up for debate if this should actually exist on the final IntentAbstraction. Depends what we do w/ Link
https://github.com/stripe/stripe-ios/blob/master/StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Extensions/Intent%2BLink.swift#L12